### ISSUE TITLE
Need ion-refresher for non-reactive functions

### ISSUE THREAD

**Issue by sferoze** [meteoric/meteor-ionic#138]
There is a need for the ion-refresher with Meteor. 

I have a Cordova functions that retrieves all available bluetooth devices. 

Each time I call this function I get an updated list of available bluetooth devices. 

When a user open the app I will call the function once to show all available bluetooth devices.

But I want to allow the user to pull down on the list to refresh by calling the function again and updating the list. 

How can I use the ion-refresher? 

Even if the ion-refresher has not been integrated is there a way I can integrate it and create a pull-request? I would need some guidance. I really need this feature. I don't want to have to include a button in the view that the user has to tap to refresh. Pull down is the expected behavior.

**Comment by nickw** [comment]
`ion-refresher` hasn't been implemented. You can either open a PR to implement it, or use an existing pull to refresh plugin:

https://github.com/jordansinger/hook.js/
https://github.com/visiongeist/pull-to-refresh-js
https://github.com/apeatling/web-pull-to-refresh/

**Comment by sferoze** [comment]
I've been trying the 3rd party libraries you listed but they don't seem to work with ionic. I might be doing something wrong? Can you verify if you have gotten them to work with ionic? 

I think the ionicView is interfering with how the 3rd party libraries implement pull to refresh. I will update if I figure out how to implement pull to refresh with your ionic package.

**Comment by sferoze** [comment]
Tried all 3 packages and could not get it to work. 

I will look into opening a PR to implement it.

**Comment by sferoze** [comment]
I think we should edit the TODO list and uncheck

-  ion-refresher (not necessary with Meteor)

**Comment by sferoze** [comment]
@nickw I want to see if I can implement ion-refresher by myself and then post a pull request to integrate into the package. 

Can you help specify the general sense in what you need to do to implement this feature. 

I looked at the code in this package a bit. Is the way is works is that you have imported all the css for the ionic framework. And then you are using Blaze templates and helper files to add and remove classes from the elements to correspond to the correct function?

So basically it is just about dropping in the native ionic css, and then just adding and removing classes at appropriate times? 

If not can you give me the general idea of how this package works to allow the ionic framework shown here: https://github.com/driftyco/ionic to be converted to this package?

**Comment by sferoze** [comment]
If you help me understand how this framework works, I will help with other components on the TODO list.

**Comment by nickw** [comment]
> Is the way is works is that you have imported all the css for the ionic framework. And then you are using Blaze templates and helper files to add and remove classes from the elements to correspond to the correct function? So basically it is just about dropping in the native ionic css, and then just adding and removing classes at appropriate times?

Typically, yes. But I believe `ion-refresher` relies on Ionic's `ion-scroll` component, which is a customized version of [Zynga Scroller](https://github.com/zynga/scroller). This allows Ionic to perform custom scrolling functionality - such as pull to refresh, among others.

So for this component we will probably need to find an off-the-shelf pull to refresh component, and then figure out how to get it to work with Ionic's CSS. This is what we did for the [`ionSideMenu`](https://github.com/meteoric/meteor-ionic/blob/master/components/ionSideMenuContainer/ionSideMenuContainer.js#L31) and [`ionSlideBox`](https://github.com/meteoric/meteor-ionic/blob/master/components/ionSlideBox/ionSlideBox.js#L11).

Here's the relevant CSS we'll want to try to use: https://github.com/meteoric/ionic-sass/blob/master/stylesheets/_scaffolding.scss#L181-L259

**Comment by sferoze** [comment]
I took a look at Zynga Scroller. Do we want to integrate this package? Should I integrate ion-scroll and then ion-refresher? since one depends on the other?

Or do you recommend finding another package?

**Comment by nickw** [comment]
@sferoze I would definitely be open to us implementing both ion-scroll and ion-refresher, but I assume that will be quite involved. If you have the bandwidth then definitely feel free to tackle this. Otherwise I was thinking an off-the-shelf plugin would be quicker.

**Comment by kylepierce** [comment]
Any update on this @sferoze? This would be a great addition to this wonderful package.

**Comment by sasikanth513** [comment]
+1 for this, this would be a great addition
